jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java

```diff
@@ -287,6 +280,8 @@ public void setIdleTimeout(int idleTimeout)
     @Override
     public void setMaxThreads(int maxThreads)
     {
+        if (maxThreads<0 || maxThreads>0x7FFF)
```

Comment:
I think it's too small to limit the QTP max threads to 32767 threads. And even 65536 seems a too small number.
I would prefer to not commit to an implementation that we will need to change as soon as we put it our because some user runs Jetty on crazy large machines.

Reply:
I can easily make a `AtomicTriInteger` with a max value of `0xFFFF` for over 1 million threads.   Note also that QTP is an easily replaceable component and anybody going over 1M threads is probably going to want a very specialized thread pool!   So will that be sufficient or you don't like the approach anyway?

Reply:
done... converted to an AtomicTriInteger with 21 bit MAX_VALUE, so it can now support 2,097,151 threads.